cpp/src/parquet/column_writer_test.cc

```diff
@@ -397,61 +404,87 @@ TYPED_TEST(TestPrimitiveWriter, RequiredRLEDictionary) {
 */
 
 TYPED_TEST(TestPrimitiveWriter, RequiredPlainWithSnappyCompression) {
-  this->TestRequiredWithSettings(Encoding::PLAIN, Compression::SNAPPY, false, false,
-                                 LARGE_SIZE);
+  this->TestRequiredWithSettings(
+      Encoding::PLAIN, Compression::SNAPPY, false, false,
+      GetCompressionCodecDefaultCompressionLevel(Compression::SNAPPY), LARGE_SIZE);
```

Comment:
Can you introduce something to avoid the duplicated instances of `GetCompressionCodecDefaultCompressionLevel(codec)`? This code diff (with this function used 12 times) suggests a needed refactor. I would suggest a struct of some kind to pass into the test function